### PR TITLE
add events-executor and timers-manager in rclcpp

### PR DESCRIPTION
**Pull request by alsora**
This PR introduces the events-executor based on the design here https://github.com/ros2/design/pull/305 and the already-existing open-source implementation in https://github.com/irobot-ros/events-executor.

FYI @clalancette @wjwwood @mjcarroll @mauropasse @bpwilcox

### CODE REVIEW

File: rclcpp/test/rclcpp/executors/test_events_executor.cpp [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+  // This second spin_once should take care of the timer,
+  executor.spin_once(10ms);
+
+  // but doesn't spin the time enough to call the timer callback.
+  EXPECT_EQ(0u, t_runs);
+  EXPECT_TRUE(std::chrono::steady_clock::now() - start < 200ms);
+}
+
+// FIX THIS TEST! The entities collector is being called too many times!
```
Comment (alsora) [inline rclcpp/test/rclcpp/executors/test_events_executor.cpp:129]: TODO: fix this test

**Comment by alsora** [comment]
Actually, given that the static executor is going to be deprecated, I was about to close https://github.com/ros2/rclcpp/pull/1891 and here I created a completely separate entities-collector.

However, I'm open to different directions.

**Comment by mjcarroll** [comment]
> here I created a completely separate entities-collector.

Let me take a look at what you have here.  I refactored the entity collection features out of the base `Executor` class because I saw it was being reproduced already similar to what you did in #1891, but with support for the `rclcpp::WaitSet`.

**Comment by alsora** [comment]
Oh ok.
Anyhow it's very easy to go back to the "split implementation".
You can find it in the irobot-events-executor repo: [base entities-collector class](https://github.com/irobot-ros/events-executor/blob/main/irobot_events_executor/src/rclcpp/executors/entities_collector_base.cpp) and [events-executor entities-collector class](https://github.com/irobot-ros/events-executor/blob/main/irobot_events_executor/src/rclcpp/executors/events_executor/events_executor_entities_collector.cpp)

the code is a direct follow up of the https://github.com/ros2/rclcpp/pull/1891 PR.

**Comment by alsora** [comment]
Disabled events-executor tests when using connext dds (events-executor support is WIP there).
Fixed some windows errors.

New CI

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18323/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12845/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18989/)

**Comment by alsora** [comment]
>  I have opened https://github.com/ros2/rclcpp/pull/2142, where I have used the EntityCollector and NotifyWaitable, can we refactor between these two implementations for common functionality?

Ok, I can review that PR and then take care of it; is it your plan to first merge that one and then we can proceed with this one?

**Comment by mjcarroll** [comment]
> is it your plan to first merge that one and then we can proceed with this one?

Maybe we can meet in the middle on the common types in another PR and then the events executor and waitset can be independent.

**Comment by alsora** [comment]
Closing this PR in favor of the new one https://github.com/ros2/rclcpp/pull/2155

**Pull request closed without merging**